query_base.go

```diff
@@ -1096,3 +1096,137 @@ func (q cascadeQuery) appendCascade(fmter schema.Formatter, b []byte) []byte {
 	}
 	return b
 }
+
+//------------------------------------------------------------------------------
+
+type idxHintsQuery struct {
+	// use index
+	uIndex           [][]schema.QueryWithArgs
+	uIndexForJoin    [][]schema.QueryWithArgs
+	uIndexForOrderBy [][]schema.QueryWithArgs
+	uIndexForGroupBy [][]schema.QueryWithArgs
+
+	// ignore index
+	ignIndex           [][]schema.QueryWithArgs
+	ignIndexForJoin    [][]schema.QueryWithArgs
+	ignIndexForOrderBy [][]schema.QueryWithArgs
+	ignIndexForGroupBy [][]schema.QueryWithArgs
```

Comment:
Let's not shorten variables like this (`u` => `use`) - it is uncommon and hard to understand.

Perhaps this should be:

```
type SelectQuery struct {
    indexHintsQuery
}

type indexHintsQuery struct {
    // lazily allocated when index is added
    use     *indexHints  
    ignore *indexHints
}

type indexHints struct {
	names         []schema.QueryWithArgs
	forJoin         []schema.QueryWithArgs
	forOrderBy  []schema.QueryWithArgs
	forGroupBy []schema.QueryWithArgs
}
```

What do you think?

Reply:
Yes, good idea
Fixed